api/proto/teleport/userintegrationtasks/v1/user_integration_tasks.proto

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.userintegrationtasks.v1;
+
+import "google/protobuf/timestamp.proto";
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/userintegrationtasks/v1;userintegrationtasksv1";
+
+// UserIntegrationTask is a resource that represents an action to be completed by the user.
+// UserIntegrationTasks are a unit of work for users to act upon issues related to integrations.
+// As an example, when auto-enrolling EC2 instances using the Discovery Service
+// a UserIntegrationTask is created to let the user know that something failed on a set of instances.
+// The user can then mark the task as resolved after following the recommendation/fixing steps.
+message UserIntegrationTask {
```

Comment:
Why is it specifically a "user integration task"? Didn't we want to make it more generic to make sure that we can use it for other types of issues as well in future?

Can we make it similar to our audit events proto, have a generic "Task" object and use OneOf to signify what type of spec it should contain, integration related or not.

Reply:
I think "task" might be too generic.
We might want to use other types of tasks for totally unrelated stuff.
This suggestion came from @espadolini

I can change it to `Task`, but I got used to UserIntegrationTask and I think it's more expressive of what it means.

Reply:
If the interaction is going to be the same for all tasks, differing only in how we display them, maybe they can be `user_task`s?

Reply:
I'm fine with UserTask 👍 
Let me know what you think @r0mant

Reply:
Renaming to UserTask